packages/block-editor/src/hooks/style.js

```diff
@@ -323,28 +305,52 @@ const withElementsStyles = createHigherOrderComponent(
 			'link'
 		);
 
-		// The Elements API only supports link colors for now,
-		// hence the specific omission of `link` in the elements styles.
-		// This might need to be refactored or removed if the Elements API
-		// changes or `link` supports styles beyond `color`.
-		const elements = skipLinkColorSerialization
-			? omit( props.attributes.style?.elements, [ 'link' ] )
-			: props.attributes.style?.elements;
-
-		const styles = compileElementsStyles(
-			blockElementsContainerIdentifier,
-			elements
-		);
+		let styles = [];
+
+		// Remove values based on whether serialization has been skipped for a specific style.
+		const rawElementsStyles = props.attributes.style?.elements;
+
+		if ( ! isEmpty( rawElementsStyles ) ) {
+			const filteredElementsStyles = {
+				...rawElementsStyles,
+				link: {
+					...rawElementsStyles?.link,
```

Comment:
I think here's a redundant optional chain.

```suggestion
					...rawElementsStyles.link,
```